components/base/sensorcontrolled/velocities.go

```diff
@@ -215,6 +226,33 @@ func (sb *sensorBase) State(ctx context.Context) ([]float64, error) {
 	return []float64{linvel.Y, angvel.Z}, nil
 }
 
+func (sb *sensorBase) validateControlLoopConfig(ctx context.Context, controlLoopConfig control.Config) error {
+	sb.blockNames = make(map[string]string)
+	hasLinConst, hasAngConst := false, false
+
+	// Verify linear and angular constant blocks exist and store their names.
+	// These two blocks are the only block names that are used by sensorBase
+	constBlocks, err := sb.loop.ConfigAtType(ctx, "constant")
+	sb.logger.CDebugf(ctx, "const blocks = %v", constBlocks)
+	if err != nil {
+		return err
+	}
+	for _, b := range constBlocks {
+		if strings.Contains(b.Name, "lin") {
+			sb.blockNames["linear_constant"] = b.Name
+			hasLinConst = true
+		} else if strings.Contains(b.Name, "ang") {
```

Comment:
This approach strikes me as brittle: if the users get to pick their own names, we're going to have trouble with a laundry bot "lint_roller_anglular_constant" or a fencing bot with a "tang_linear_constant". This gets even worse for someone who doesn't speak English and has named things in their native language.

These blocks are required for the component, yeah? Can we put their names in the config? Like,
```
type BaseConfig struct {
    LinearBlockName string
    AngularBlockName string
}
```
That way, you don't need to make assumptions about the names at all: you just get handed the right ones.

Reply:
the problem is that the only differentiation between the two blocks is what they are named. there is really no other way to tell which is which so I don't think I understand how we can directly add them to the config. as of right now we are building in the control config anyways, so we are picking the names of the blocks and it doesn't really matter, but in a future case where we allow users to add their own control config it will be an issue

Reply:
I don't understand how we can get the names in the way you are suggesting, but this gave me a new idea to use the depends on field in some way, so I am going to try that

Reply:
just kidding constants don't depend on anything so that's the biggest issue

Reply:
Hm... I suspect the problem is that I don't really know what I'm talking about and have misunderstood something I don't know that I don't know. When convent, I'd like to understand the very basics of how to set up blocks, how to connect them together, and why to use this design (which seems more open-ended and customizable than I know what to do with).

I imagine that after that, it will be obvious that you've made the right choices in here and I just didn't see it before.

Reply:
I should be back in person on Monday and we can discuss in person!